cmake/Platform/WASI.cmake

```diff
@@ -1 +1,2 @@
 set(WASI 1)
+set(CMAKE_EXECUTABLE_SUFFIX ".wasm")
```

Comment:
That's a good point yeah and the intention of this is to affect the cmake integration of wasi yeah. I'm not sure how widely used it is and to what extent but without this binaries would show up as `foo` instead of `foo.wasm` which wasn't what the test suite was expecting at least. That being said I'm happy to move this into just the test suite for now.